processor/resourcedetectionprocessor/internal/aws/elasticbeanstalk/elasticbeanstalk.go

```diff
@@ -0,0 +1,83 @@
+// Copyright -c OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticbeanstalk
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	TypeStr     = "elasticBeanstalk"
+	linuxPath   = "/var/elasticbeanstalk/xray/environment.conf"
+	windowsPath = "C:\\Program Files\\Amazon\\XRay\\environment.conf"
+)
+
+var _ internal.Detector = (*Detector)(nil)
+
+type Detector struct {
+	fs readOnlyFileSystem
+}
+
+type EbMetaData struct {
+	DeploymentID    int    `json:"deployment_id"`
+	EnvironmentName string `json:"environment_name"`
+	VersionLabel    string `json:"version_label"`
+}
+
+func NewDetector() (internal.Detector, error) {
+	return &Detector{fs: &ebReadOnlyFileSystem{}}, nil
+}
+
+func (d Detector) Detect(context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	var conf file
+	var err error
+
+	if d.fs.IsWindows() {
+		conf, err = d.fs.Open(windowsPath)
+	} else {
+		conf, err = d.fs.Open(linuxPath)
+	}
+
+	if err != nil {
+		log.Println("Unable to find Elastic Beanstalk config file, skipping resource detection")
+		return res, err
+	}
+
+	ebmd := &EbMetaData{}
+	err = json.NewDecoder(conf).Decode(ebmd)
+	defer conf.Close()
```

Comment:
Would either put the defer right after the err checking or since it looks like you're done reading the file here just `conf.Close()` with no defer?

Reply:
Good catch don't think a defer is necessary, this was just a bit too mindless of a copy-paste from an online example.